src/middlewared/middlewared/main.py

```diff
@@ -106,12 +105,28 @@ def register_callback(self, name, method):
         self.__callbacks[name].append(method)
 
     def _send(self, data):
+        serialized = json.dumps(data)
+        asyncio.run_coroutine_threadsafe(self.response.send_str(serialized), loop=self.loop)
+        _1MB = 1000000
+        if sys.getsizeof(serialized) > _1MB:
```

Comment:
why sys.getsizeof? Its a string, can use len()?

Reply:
Also 1MB seems excessive, I would limit it to 10KiB or something? what else would use such a big message?

Reply:
```
>>> zz = 'qweoirupqweriopkjlasdf;klasdklfjpouiqweriuoqwiueorask;jldf;lkjasdfasdklff'
>>> len(zz)
73
>>> sys.getsizeof(zz)
122
```

Reply:
Since its a string is just a few extra bytes for the object, negligible